### ISSUE TITLE
Accept authentication via remote_user header

### ISSUE THREAD

**Issue by LaramieSmile** [opendistro-for-elasticsearch/community#87]
Hey All,
I was wondering if there was a way, or if it could be a feature request, to allow using the built in HTTP authorization for HTTP access based on a passed REMOTE_USER header.

Use case is I already have authentication setup in front of kibana(proxied by Apache) using our orgs kerberos and 2FA authentication.  I'd rather be able to keep the 2FA and just pass the user to Kibana/elastic and have it use the built in HTTP roles backend.

**Comment by jimmyjones2** [comment]
Yes, but for some reason isn't in the docs - I think this needs fixing. Unless they're planning on removing it?!

There is something in the config, or read the docs where the code originally came from: https://docs.search-guard.com/latest/proxy-authentication
